akka-projection-grpc/src/main/scala/akka/projection/grpc/internal/FilterStage.scala

```diff
@@ -88,7 +86,8 @@ import akka.stream.stage.StageLogging
     sliceRange: Range,
     var initFilter: Iterable[FilterCriteria],
     currentEventsByPersistenceIdQuery: CurrentEventsByPersistenceIdQuery,
-    verbose: Boolean = false)
+    verbose: Boolean = false,
+    val envelopeFilter: EventEnvelope[Any] => Boolean = _ => false)
```

Comment:
can we come up with a better name than `envelopeFilter`? `producerDefinedFilter` ?

Reply:
default `false` will filter out all? probably best to remove the default value of this parameter

Reply:
There could be alternative ways to see this. I think the producer filter should have higher authority than the consumer filter because the producer owns the data. If the producer filter says no I don't think there should be a way to override that by the consumer filter. In other words `producerFilter(env) && consumerFilter.matches(pid)`.

Reply:
Switched over in [bb5a57e](https://github.com/akka/akka-projection/pull/819/commits/bb5a57ea7e31c5ad346940fc3b96b536cb6a6db1)